### PR TITLE
Finishlinelab3

### PR DESCRIPTION
**Pull request by ngengecharity**

### CODE REVIEW

**Comment by ngengecharity** [comment]
error

**Comment by ngengecharity** [comment]
**Summary:**
Modified Cloudformation

File: Finishlinelab3/finishlinelab3infra.yaml [hunk 1]
```diff
@@ -29,17 +30,18 @@ Parameters:
...
+  CharityWebServer:
     Type: 'AWS::EC2::Instance'
     Properties:
       InstanceType: !Ref InstanceType
       SecurityGroups:
         - !Ref InstanceSecurityGroup
       KeyName: !Ref KeyName
       ImageId: !Ref AmiId 
-      Tag:
-        - Name: !Ref CharityInstance
```
Comment (ndzohdong) [inline Finishlinelab3/finishlinelab3infra.yaml:41]: Wrong syntax Tag . It should look like this
```
Tags:
   - Key: Name
   - Value: CharitywebServer
```

File: Finishlinelab3/finishlinelab3infra.yaml [hunk 2]
```diff
@@ -48,6 +50,26 @@ Resources:
           FromPort: '22'
           ToPort: '22'
           CidrIp: !Ref SSHLocation
```
Comment (ndzohdong) [inline Finishlinelab3/finishlinelab3infra.yaml]: missing web port . Please add http ingress rule below
`- IpProtocol: tcp
         FromPort: 80
         ToPort: 80
         CidrIp: !Ref 'SSHLocation'`

File: Finishlinelab3/finishlinelab3infra.yaml [hunk 3]
```diff
@@ -48,6 +50,26 @@ Resources:
...
+  CharityDBServer:
+    Type: 'AWS::EC2::Instance'
+    Properties:
+      InstanceType: !Ref InstanceType
+      SecurityGroups:
+        - !Ref InstanceSecurityGroup
```
Comment (ndzohdong) [inline Finishlinelab3/finishlinelab3infra.yaml]: You will have to reference both Web SG and DB SG

` - !Ref CharityWebServerSG
  - !Ref CharityDBSG`

File: Finishlinelab3/finishlinelab3infra.yaml [hunk 4]
```diff
@@ -57,8 +79,8 @@ Outputs:
     Value: !GetAtt 
       - CharityInstance
       - AvailabilityZone
-  PublicIP:
```
Comment (ndzohdong) [inline Finishlinelab3/finishlinelab3infra.yaml:60]: Poor indentation should be at the same level as CharityInstance

File: Finishlinelab3/finishlinelab3infra.yaml [hunk 5]
```diff
@@ -57,8 +79,8 @@ Outputs:
...
-    Description: Public IP address of the newly created EC2 instance
+  PublicDNS:
+    Description: Public DNSName of the newly created EC2 instance
     Value: !GetAtt 
       - CharityInstance
```
Comment (ndzohdong) [inline Finishlinelab3/finishlinelab3infra.yaml]: This should be the name of your webserver

**Review by ndzohdong — CHANGES_REQUESTED** [review]

File: Finishlinelab3/finishlinelab3infra.yaml (hunk 3 above)
Comment (ngengecharity) [reply Finishlinelab3/finishlinelab3infra.yaml]: done

File: Finishlinelab3/finishlinelab3infra.yaml (hunk 5 above)
Comment (ngengecharity) [reply Finishlinelab3/finishlinelab3infra.yaml]: done

File: Finishlinelab3/finishlinelab3infra.yaml (hunk 3 above)
Comment (ngengecharity) [reply Finishlinelab3/finishlinelab3infra.yaml]: done

File: Finishlinelab3/finishlinelab3infra.yaml (hunk 2 above)
Comment (ngengecharity) [reply Finishlinelab3/finishlinelab3infra.yaml]: done

**Review by ndzohdong — APPROVED** [review]

**Pull request merged**